lib/Parser/Parse.cpp

```diff
@@ -4609,6 +4609,12 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
         isEnclosedInParamScope = this->m_currentScope->GetEnclosingScope() && this->m_currentScope->GetEnclosingScope()->GetScopeType() == ScopeType_Parameter;
     }
 
+    if (isEnclosedInParamScope)
+    {
+        // We cannot just do HasDefaultArguments here becasue we may be inside a destructuring pattern.
+        Assert(pnodeFncParent && pnodeFncParent->sxFnc.HasNonSimpleParameterList());
+    }
```

Comment:
Agreed - also there is no need for a comment though. can pnodeFncParent be null in any case?

---
In reply to: [53990567](https://github.com/Microsoft/ChakraCore/pull/384#discussion_r53990567) [](ancestors = 53990567)